src/Infrastructure/Captcha/Startup.cs

```diff
@@ -0,0 +1,28 @@
+using Microsoft.AspNetCore.Builder;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.DependencyInjection;
+
+namespace FSH.WebApi.Infrastructure.Captcha;
+
+internal static class Startup
+{
+    internal static IServiceCollection AddCaptcha(this IServiceCollection services, IConfiguration config)
+    {
+        var settings = config.GetSection(nameof(CaptchaSettings)).Get<CaptchaSettings>();
+
+        if (string.IsNullOrEmpty(settings?.UseCaptcha) is false && settings.UseCaptcha == nameof(GoogleRecaptchaV3))
+        {
+            if (string.IsNullOrWhiteSpace(settings.GoogleRecaptchaV3?.Secret) ||
+                string.IsNullOrWhiteSpace(settings.GoogleRecaptchaV3?.ApiUrl))
+            {
+                // logger
+                throw new Exception("Invalid 'Secret' or 'Url' properties in captcha.json. Parent: GoogleRecaptchaV3.");
+            }
+
+            services.AddHttpClient<ICaptchaService, GoogleRecaptchaV3Service>();
+            services.AddTransient<ICaptchaService, GoogleRecaptchaV3Service>();
```

Comment:
Here I mean, you can specify the base url...
Also not sure if that AddTransient is actually necessary if you registered it as an httpclient already...

Reply:
AddHttpClient should be enough for the implementation

Reply:
both is true..
but
in docs and samples of httpclient usage 
at start of app defining httpclient if i remember truely.

base url sometimes is a headache. if you use morethen one you have to manage it or multiply httpclient  service in container.